spring-context/src/main/java/org/springframework/context/i18n/LocaleContextThreadLocalAccessor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.context.i18n;
+
+import io.micrometer.context.ThreadLocalAccessor;
+
+/**
+ * A {@link ThreadLocalAccessor} to put and restore the current {@link LocaleContext}.
+ *
+ * @author Tadaya Tsuyukubo
+ * @since 6.2.0
+ */
+public class LocaleContextThreadLocalAccessor implements ThreadLocalAccessor<LocaleContext> {
+
+	/**
+	 * Key under which this accessor is registered in
+	 * {@link io.micrometer.context.ContextRegistry}.
+	 */
+	public static final String KEY = "spring.framework.locale-context";
```

Comment:
We tend to use something like `LocaleContextThreadLocalAccessor.class.getName()` + ".KEY".